### DISCUSSION TITLE
MicroSoft Access development

### DISCUSSION THREAD

**Discussion by hmetz** [community/community#138054] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have been working on a MS Access project of 20 years and just realized get open source help makes sense.
My question is whether oss makes sense for projects using MS Access fit.

**Comment by VadymMakohon** [comment]
Using open-source software (OSS) methodologies for a project that relies on Microsoft Access can make sense in certain contexts, but there are a few factors to consider. MS Access itself is proprietary software, and there are limitations to how it interacts with open-source development practices. However, depending on your goals, open-sourcing the logic and structure of your project could still be beneficial.

When OSS makes sense for an MS Access project:
Sharing Logic and Documentation:

You can share the database design, queries, VBA code, and macros used in your MS Access project as open-source. This can allow the broader community to contribute improvements, provide feedback, and offer support.
You can release the non-Access components of your project (such as scripts, documentation, or interfaces) under an open-source license.
Learning from the Community:

By making parts of your project open-source, you can receive help from the community in terms of optimization, finding bugs, and offering new features. This is particularly useful if you plan to migrate or integrate your MS Access database with other open technologies in the future.
Migration to More Open Platforms:

If you are considering moving your MS Access project to a more open database platform (like MySQL, PostgreSQL, or SQLite), open-sourcing the project can make this process smoother. The community could help you build migration tools, convert VBA logic to other programming languages, or create web-based alternatives.
When OSS may not be the best fit:
Proprietary Nature of MS Access:

MS Access itself is a closed-source, proprietary platform. Anyone contributing to or using your project would need a licensed version of MS Access, which can limit participation.
Open-source developers might prefer using free and open databases like MySQL or PostgreSQL, which makes it harder to find contributors familiar with Access.
Legacy Dependencies:

If your project relies heavily on MS Access-specific features or integrates with other Microsoft products (like Excel or Outlook), moving it to an open-source environment may be challenging. Some features in Access may not be easily replicated in other database platforms.
Potential Path Forward:
Open-source the logic: You could release the VBA code or SQL logic used in your MS Access project, allowing others to contribute and adapt it for other platforms.
Consider migration: If you're thinking of future-proofing your project, consider migrating to an open-source database. In that case, open-sourcing the process can invite community help to ease the migration.

**Comment by rioyuda02** [comment]
Open-source solutions can certainly complement or even replace certain aspects of a project that uses MS Access, but there are a few considerations to keep in mind when evaluating whether this approach makes sense:

Platform Compatibility: MS Access is a proprietary Microsoft product, and while it offers great integration with other Microsoft tools, it is less suited to cross-platform environments. If you want to move towards open-source, you'd likely need to consider migrating to a database system like MySQL, PostgreSQL, or SQLite, which are open-source alternatives.

Database Complexity: If your MS Access project has complex relationships, queries, forms, and reports, replicating this functionality with open-source tools can be a significant effort. You’d likely need a front-end replacement (such as a web app or open-source GUI tool) to mimic Access’s interface.

Cost and Control: Open-source tools are free to use and provide greater control over customization, scaling, and security. If your project is now constrained by the licensing or limitations of MS Access, switching to open-source could help.

Migration Tools: Some tools, such as LibreOffice Base, or migration services, can help transition an Access database into a more open environment. However, depending on the complexity of your project, this could require extensive manual effort.

Community Support: Open-source communities can provide a wealth of free support, which might complement the work you’ve been doing with MS Access, especially for adding new features, optimizing performance, or troubleshooting.

If your project is heavily reliant on Access's GUI features, you might need to weigh whether the transition to open-source (and the possible need to develop a new interface) is worth the investment. But if the goal is database flexibility, scalability, and long-term support, migrating part of the system to open-source could make a lot of sense.

**Reply by hmetz** [reply]
Thank you. It is clear that until I decide to reengineer using another tool Github does not make sense.

I am pretty good with databases so I am using MS Access for the front end.
 What tool would you recommend for building a front end?
